### PR TITLE
Fix Bug 1432899 - Remove Top Sites edit modal

### PR DESCRIPTION
**Pull request by rlr**
lots of red! yay

### CODE REVIEW

File: system-addon/content-src/components/TopSites/TopSite.jsx [hunk 1]
```diff
@@ -157,76 +149,25 @@ export class TopSite extends React.PureComponent {
...
-              onClick={this.onEditButtonClick} />
-            <button
-              className="icon icon-dismiss"
-              title={this.props.intl.formatMessage({id: "edit_topsites_dismiss_button"})}
-              onClick={this.onDismissButtonClick} />
-          </div>
-        }
+        <div>
+          <button className="context-menu-button icon" onClick={this.onMenuButtonClick}>
+            <span className="sr-only">{`Open context menu for ${title}`}</span>
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSite.jsx]: Hmmm.... I see this was added way back in #2547 but this should probably be a localized string… @sarracini ?

File: system-addon/content-src/components/TopSites/TopSite.jsx [hunk 2]
```diff
@@ -157,76 +149,25 @@ export class TopSite extends React.PureComponent {
   onMenuUpdate(showContextMenu) {
     this.setState({showContextMenu});
   }
-  onDismissButtonClick() {
-    const {link} = this.props;
-    if (link.isPinned) {
-      this.props.dispatch(ac.SendToMain({
-        type: at.TOP_SITES_UNPIN,
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSite.jsx]: Odd this was here to begin with as blocking will unpin https://searchfox.org/mozilla-central/source/toolkit/modules/NewTabUtils.jsm#556

File: locales/en-US/strings.properties [hunk 3]
```diff
@@ -136,6 +136,7 @@ edit_topsites_unpin_button=Unpin this site
 edit_topsites_edit_button=Edit this site
```
Comment (Mardak) [inline locales/en-US/strings.properties]: Hmm... let's try adding a plain `# remove after 60` comment before each string that's now unused

File: system-addon/content-src/components/TopSites/TopSites.jsx [hunk 4]
```diff
@@ -40,6 +40,17 @@ function countTopSitesIconsTypes(topSites) {
 }
 
 export class _TopSites extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = this.DEFAULT_STATE = {
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSites.jsx]: This multi assign is disallowed by eslint now. See this for an example of how to fix:
https://github.com/mozilla/activity-stream/pull/3948/commits/1f8682475fa4a3fc3c12f597e172e496598db805#diff-8a848a318e8cc982dc70dfd24807f0ef

File: system-addon/content-src/components/TopSites/TopSites.jsx [hunk 5]
```diff
@@ -60,25 +71,77 @@ export class _TopSites extends React.PureComponent {
...
   }
-
+  onAddButtonClick() {
+    this.setState({showAddForm: true});
+    this.props.dispatch(ac.UserEvent({
+      source: TOP_SITES_SOURCE,
+      event: "TOP_SITES_ADD_FORM_OPEN"
+    }));
+  }
+  onFormClose() {
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSites.jsx]: There'll need to be spaces when rebased with latest eslint rules

File: system-addon/test/functional/mochitest/browser_topsites_section.js [hunk 6]
```diff
@@ -2,13 +2,13 @@
 
 // Check TopSites edit modal and overlay show up.
 test_newtab(
-  // it should be able to click the topsites edit button to reveal the edit topsites modal and overlay.
+  // it should be able to click the topsites add button to reveal the add top site modal and overlay.
   function topsites_edit() {
-    const topsitesEditBtn = content.document.querySelector(".edit-topsites-button button");
+    const topsitesEditBtn = content.document.querySelector(".add-topsites-button button");
     topsitesEditBtn.click();
```
Comment (Mardak) [inline system-addon/test/functional/mochitest/browser_topsites_section.js]: let's fix up the variable name too

**Comment by as-pine-proxy** [comment]
[Treeherder: caf24e52d6652f16719dedce4b32ea78fb583937](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=caf24e52d6652f16719dedce4b32ea78fb583937)

**Pull request merged**